examples/rust/external_data_loader/src/main.rs

```diff
@@ -56,18 +75,52 @@ fn main() -> anyhow::Result<()> {
 
     let rec = {
         let mut rec = rerun::RecordingStreamBuilder::new("rerun_example_external_data_loader");
-        if let Some(recording_id) = args.recording_id {
+        if let Some(recording_id) = args.recording_id.as_ref() {
             rec = rec.recording_id(recording_id);
         };
 
         // The most important part of this: log to standard output so the Rerun Viewer can ingest it!
         rec.stdout()?
     };
 
-    rec.log_timeless(
-        rerun::EntityPath::from_file_path(&args.filepath),
+    // TODO(#3841): really need those send APIs
+    rec.set_timepoint(timepoint_from_args(&args)?);
+
+    let entity_path_prefix = args
+        .entity_path_prefix
+        .map_or_else(|| rerun::EntityPath::new(vec![]), rerun::EntityPath::from);
+
+    rec.log_with_timeless(
+        entity_path_prefix.join(&rerun::EntityPath::from_file_path(&args.filepath)),
+        args.timeless,
         &rerun::TextDocument::new(text).with_media_type(MediaType::MARKDOWN),
     )?;
 
     Ok::<_, anyhow::Error>(())
 }
+
+fn timepoint_from_args(args: &Args) -> anyhow::Result<rerun::TimePoint> {
+    let mut timepoint = rerun::TimePoint::default();
+    {
+        for time_str in &args.time {
+            let Some((timeline_name, time)) = time_str.split_once('=') else {
+                continue;
+            };
+            timepoint.insert(
+                rerun::Timeline::new_temporal(timeline_name),
+                time.parse::<i64>()?.into(),
+            );
+        }
+
+        for seq_str in &args.sequence {
+            let Some((seqline_name, seq)) = seq_str.split_once('=') else {
+                continue;
+            };
+            timepoint.insert(
+                rerun::Timeline::new_sequence(seqline_name),
+                seq.parse::<i64>()?.into(),
+            );
+        }
+    }
+    Ok(timepoint)
+}
```

Comment:
feels like somthing we may want to provide as a public utility? :thinking:

Reply:
I'd like to keep the whole external `DataLoader` API surface as small as possible for now -- we know we're going to break things a million times; I'd rather break less...